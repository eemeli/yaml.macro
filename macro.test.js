const pluginTester = require('babel-plugin-tester').default
const plugin = require('babel-plugin-macros')

pluginTester({
  plugin,
  babelOptions: { filename: __filename },
  tests: {
    // happy path
    'flat collection': {
      code: `
        import yaml from './macro'
        const foo = yaml('./__fixtures__/flat-collection.yaml')
      `,
      snapshot: true
    },
    'deep collection': {
      code: `
        import yaml from './macro'
        const foo = yaml('./__fixtures__/deep-collection.yaml')
      `,
      snapshot: true
    },
    sequence: {
      code: `
        import yaml from './macro'
        const foo = yaml('./__fixtures__/sequence.yaml')
      `,
      snapshot: true
    },
    number: {
      code: `
        import yaml from './macro'
        const foo = yaml('./__fixtures__/number.yaml')
      `,
      snapshot: true
    },
    string: {
      code: `
        import yaml from './macro'
        const foo = yaml('./__fixtures__/string.yaml')
      `,
      snapshot: true
    },

    // argument evaluation
    'concatenated filename': {
      code: `
        import yaml from './macro'
        const dir = './__fixtures__'
        const foo = yaml(dir + '/number.yaml')
      `,
      output: "const dir = './__fixtures__'\nconst foo = 42"
    },
    'template literal filename': {
      code: `
        import yaml from './macro'
        const dir = "__fixtures__"
        const foo = yaml(\`\${'./' + dir}/number.yaml\`)
      `,
      output: "const dir = '__fixtures__'\nconst foo = 42"
    },
    'no arguments': {
      code: `
        import yaml from './macro'
        const foo = yaml()
      `,
      error: 'yaml.macro argument evaluation failed'
    },

    // options
    'YAML options': {
      code: `
        import yaml from './macro'
        const foo = yaml('./__fixtures__/number.yaml', { schema: 'failsafe' })
      `,
      output: "const foo = '42'"
    },
    'variable schema': {
      code: `
        import yaml from './macro'
        const schema = "failsafe"
        const foo = yaml('./__fixtures__/number.yaml', { schema })
      `,
      output: "const schema = 'failsafe'\nconst foo = '42'"
    },
    'generated options object': {
      code: `
        import yaml from './macro'
        const opt = { schema: "failsafe" }
        const foo = yaml('./__fixtures__/number.yaml', opt)
      `,
      output: "const opt = {\n  schema: 'failsafe'\n}\nconst foo = '42'",
    },

    // errors
    'bad usage': {
      code: `
        import yaml from './macro'
        const foo = yaml
      `,
      error: 'yaml.macro only supports usage as a function call'
    },
    'file not found': {
      code: `
        import yaml from './macro'
        const foo = yaml('./does-not-exist.yaml')
      `,
      error: 'Cannot resolve module'
    },
    'YAML parse error': {
      code: `
        import yaml from './macro'
        const foo = yaml('./__fixtures__/bad-collection.yaml')
      `,
      error: 'Nested mappings are not allowed in compact mappings'
    }
  }
})
