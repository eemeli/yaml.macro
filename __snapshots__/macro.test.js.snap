// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`unknown plugin deep collection: deep collection 1`] = `

import yaml from './macro'
const foo = yaml('./__fixtures__/deep-collection.yaml')

      ↓ ↓ ↓ ↓ ↓ ↓

const foo = {
  first: {
    second: [
      'seq',
      {
        deep: true
      }
    ],
    third: 'value'
  },
  other: true
}


`;

exports[`unknown plugin flat collection: flat collection 1`] = `

import yaml from './macro'
const foo = yaml('./__fixtures__/flat-collection.yaml')

      ↓ ↓ ↓ ↓ ↓ ↓

const foo = {
  foo: 'bar',
  one: 1
}


`;

exports[`unknown plugin number: number 1`] = `

import yaml from './macro'
const foo = yaml('./__fixtures__/number.yaml')

      ↓ ↓ ↓ ↓ ↓ ↓

const foo = 42


`;

exports[`unknown plugin sequence: sequence 1`] = `

import yaml from './macro'
const foo = yaml('./__fixtures__/sequence.yaml')

      ↓ ↓ ↓ ↓ ↓ ↓

const foo = ['one', ['2.1', '2.2'], 'three', 4]


`;

exports[`unknown plugin string: string 1`] = `

import yaml from './macro'
const foo = yaml('./__fixtures__/string.yaml')

      ↓ ↓ ↓ ↓ ↓ ↓

const foo =
  'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.\\n'


`;
